### ISSUE TITLE
Placement not working from theme or backend

### ISSUE THREAD

**Issue by Flarescape** [OrchardCMS/OrchardCore#11840]
I tried to place a custom part inside a widget in the admin UI via placement.json in my theme and also via the placement editor in the backend.
Neither of them did anything and i mean they behave like they don't exist.

I read every bit of documentation, watched the placement videos on youtube, looked into the drive and placement of the SEO module, everything and maybe i'm just to dumb for this, but it seems as something is not working as it should and it's not just me.

My part is namend "Spacing" which has 4 text fields.
I have a content type named "Container" with stereotype "Widget", which gets the Spacing-Part attached.
Now i want to display the Spacing-Part in a tab inside the widget.

Therefore i wrote this placement json:
```json
{
  "Spacing_Edit": [
    {
      "place": "Parts#Spacing;40"
    }
  ]
}
```
I also tried this:
```json
{
  "Spacing_Edit": [
    {
      "place": "Parts#Spacing;40",
      "contentType": "Container"
    }
  ]
}
```

And this:
```json
{
  "Spacing_Edit": [
    {
      "place": "Parts:0#Spacing;1",
      "contentType": "Container"
    }
  ]
}
```

And all other possible variations of this.
Absolutely nothing happens, also not after restarting the site in VS or whatever.
It still looks exactly like this:
![image](https://user-images.githubusercontent.com/1375119/172906538-ee298e4e-0521-4b2d-ba78-bb1129a3919a.png)

Can somebody please describe to me how this is supposed to work or test if it actually does?
Current OC Version 1.4.0

**Comment by ns8482e** [comment]
Try
```

{
  "ContentPart_Edit": [
    {
      "place": "Parts#Spacing;40",
       “contentPart”: [“Spacing”]
    }
  ]
}

```

**Comment by Flarescape** [comment]
@ns8482e Thanks for your answer, it did...something, but it's even more strange now.
When i add yours to the placement.json in the theme, it does nothing, when i add it as placement in the backend, the following happens:

I added the part to the page itself and to the container widget, just for testing.
Now everything from the content type page is inside the "Spacing" tab, but inside the container, where i want just the 4 fields inside this tab, nothing happens.
The placement thing is really bad documented in my opinion.
Any ideas please?

![image](https://user-images.githubusercontent.com/1375119/173069168-e20a5a0e-51f8-4fbc-b88e-b76023964565.png)

**Comment by ns8482e** [comment]
Placement only works on shapes in scope of current theme.  

Here ContentPart_Edit shape that you are trying to place is rendered in scope of TheAdmin theme so placement defined in placement.json of the front end theme doesn’t apply but works when you define using placement feature in admin ui.

`contentPart` placement rule is applied to all of `contentPart` defined in any `ContentType` to target specific content type you can also use `contentPart` rule or  `differentiator`

**Comment by Flarescape** [comment]
Ok, the thing with the front end theme makes sense, i'm sorry, i could've thought about that.
Still i don't really get the whole contentType, contentPart and differentiator thing.
I added your placement rule inside the backend and it looked like the picture above, but i already targeted contentPart Spacing, so why are all the other parts of Page also in that tab, but more relevant, how to just place the 4 fields of the part inside the container widget in a Spacing-Tab?
The goal is to just target the container widget or any other widget i add the Spacing-Part to and therefore create a Spacing-Tab inside this widget with just the 4 fields of this part.

I'm sorry if i miss the point, but it's not really understandable to me, how to figure this out.
I've added a differentiator for example and now nothing is working again, but OC doesn't provide any infos about this behavior.

**Comment by Flarescape** [comment]
@ns8482e GitHub is displaying it wrong, please look in the "Edit" at the most recent, when you habe time, thank you.

**Comment by Flarescape** [comment]
It works if i target every single field individually, but not the whole part....that's really annoying.

**Comment by Flarescape** [comment]
@ns8482e  Thank you, that helped, even if i really don't know how somebody should figure that out by themselfes, because the docs don't really explain that well to not.
I mean, i work with Orchard for 9 to 10 years now and sadly never figured that really out.

Just one last question please, because i'm still just guessing around how to write that with placement, but i have the Spacing-Part attached to the Container-Content-Type with stereotype "Widget", to use it in Flow Part, which looks like this now:
![image](https://user-images.githubusercontent.com/1375119/173108257-f177f7f7-27b5-4240-b10b-f0e4cdc8f051.png)

How to define the Spacing-Part now as a Wrapper for Container?
The Container-Template is just:
`{{ Model.Content | shape_render }}`

And the Spacer-Part-Template is:
```html
<div class="wrapper">
{{ Model.Content | shape_render }}
</div>
```

I tried the following (which of course doesn't work, because i have really no idea):
{
  "ContentType": [
    {
      "place": "Content:1",
      "differentiator": "Container",
      "contentType": ["Container"],
      "wrappers": ["Spacing"]
     }
  ]
}

**Comment by ns8482e** [comment]
If I understand correctly you don’t want to render ootb template of SpacingPart inside container content type instead you want to render a wrapper around container that uses data of SpacingPart correct?

- you can hide SpacingPart using template
- You can create a SpacingWrapper shape and set it as wrapper to container using placement
- Inside SpacingWrapper you can read part data using Model.ContentItem.Content.SpacingPart

**Comment by Flarescape** [comment]
@ns8482e Correct, i want to do a Spacing-Wrapper for Container with the data from the Spacing-Part that is attached to the container, i just don't know how to do the placement for the wrapper and i tried a lot!
It's just not really clear to me how the json should look like, i'm sorry!

**Comment by ns8482e** [comment]
@Flarescape I'll get some time today and will share some sample code

**Comment by ns8482e** [comment]
> • you can hide SpacingPart using placement 
```
"ContentPart": [
    {
      "place": "-",
      "differentiator": "Spacing",
      "contentType": [ "Container" ]
    }
  ] 
```
> • You can create a Spacing Wrapper shape and set it as wrapper to container 

**Widget-Container.liquid**
```liquid
{% shape_add_wrappers Model "Animation Spacer" %}
{{ Model.Content | shape_render }}
``` 

> • Inside Spacing Wrapper you can read part data using Model.ContentItem.Content.SpacingPart 

**Spacer.liquid**
```liquid
{% assign top = Model.ContentItem.Content.SpacingPart.Top%}
<div style='margin-top: {{top}}'>
 {{ Model.Metadata.ChildContent }}
</div>
```

**Animation.liquid**
` {{ Model.Metadata.ChildContent }}`

**Comment by sebastienros** [comment]
Hope we can make this a documentation page when it's resolved.

**Comment by tripurarisingh1** [comment]
Remember, contributions to this repository should follow its contributing guidelines, security policy, and code of conduct.
